File: Python/terminal.py
from sys import stdout


def goto_xy(x: int, y: int) -> None:
    stdout.write(f"\033[{y};{x}H")
    stdout.flush()


def hide_cursor() -> None:
    stdout.write("\033[?25l")
    stdout.flush()


def show_cursor() -> None:
    stdout.write("\033[?25h")
    stdout.flush()


def clear_display() -> None:
    stdout.write("\033[2J\033[H")
    stdout.flush()

File: Rust/src/terminal.rs
use std::io::{self, Write};

pub fn goto_xy(x: i32, y: i32) {
    print!("\x1B[{};{}H", y, x);
    let _ = io::stdout().flush();
}

pub fn hide_cursor() {
    print!("\x1B[?25l");
    let _ = io::stdout().flush();
}

pub fn show_cursor() {
    print!("\x1B[?25h");
    let _ = io::stdout().flush();
}

pub fn clear_display() {
    print!("\x1B[2J\x1B[H");
    let _ = io::stdout().flush();
}
